src/routing/approved.rs

```diff
@@ -1621,33 +1621,41 @@ impl Approved {
     }
 
     fn send_sync(&mut self, section: Section, network: Network) -> Result<Vec<Command>> {
-        let mut commands = vec![];
-
-        for peer in section.active_members() {
-            if peer.name() == &self.node.name() {
-                continue;
-            }
-
-            let variant = if section.is_elder(peer.name()) {
-                Variant::Sync {
-                    section: section.clone(),
-                    network: network.clone(),
-                }
-            } else {
-                Variant::Sync {
-                    section: section.to_minimal(),
-                    network: Network::new(),
-                }
-            };
+        let send = |variant, recipients: Vec<_>| -> Result<_> {
+            trace!("Send {:?} to {:?}", variant, recipients);
 
-            trace!("Send {:?} to {:?}", variant, peer);
             let message =
                 Message::single_src(&self.node, DstLocation::Direct, variant, None, None)?;
-            commands.push(Command::send_message_to_target(
-                peer.addr(),
+            let recipients: Vec<_> = recipients.iter().map(Peer::addr).copied().collect();
+
+            Ok(Command::send_message_to_targets(
+                &recipients,
+                recipients.len(),
                 message.to_bytes(),
             ))
-        }
+        };
+
+        let mut commands = vec![];
+
+        let (elders, non_elders): (Vec<_>, _) = section
+            .active_members()
+            .filter(|peer| peer.name() != &self.node.name())
+            .copied()
+            .partition(|peer| section.is_elder(peer.name()));
+
+        // Send the trimmed state to non-elders. The trimmed state contains only the latest
+        // section key and one key before that which is the key the recipients should know so they
```

Comment:
Yes, when we append a new section key to the chain, the members need to receive a chain that contains the key they already know and the new key so that they can verify the new key. Under normal circumstances, the key they already know is the one before the latest key. In case they are lagging and their latest known key is further behind, they will bounce the message and it will be resent back to them with extended chain. The bouncing of the `Sync` messages were not handled correctly which is what this PR fixes.